cmd/snap/cmd_prepare_image.go

```diff
@@ -49,8 +50,17 @@ func init() {
 }
 
 func (x *cmdPrepareImage) Execute(args []string) error {
+	var modelFile, modelRef string
+
+	if osutil.FileExists(x.Positional.ModelAssertion) {
```

Comment:
Having a completely different mode of operation based solely on the existence of a provided parameter as a file is often a bad idea. In the best case, it means a simple typo changes the operation into something broken in surprising ways, and in the worst case it introduces security bugs.

I'll suggest a different approach below, based on another factor.